protocol/contracts/governance/ElderElection.sol

```diff
@@ -66,23 +69,29 @@ contract ElderElection is Ownable, EIP712 {
      */
     function setNominationFee(uint256 _nominationFee) external onlyOwner {
         nominationFee = _nominationFee;
+        emit UpdateNominationFee(_nominationFee);
     }
 
     /**
-     * @notice Update the merkle tree root for temple roles.
+     * @notice Update the hash root for temple roles.
      */
-    function setTemplarRolesMerklRoot(bytes32 _templarRolesMerkleRoot) external onlyOwner {
-        templarRolesMerkleRoot = _templarRolesMerkleRoot;
+    function setTemplarRolesMerklRoot(bytes32 _templarRolesHash) external onlyOwner {
+        templarRolesHash = _templarRolesHash;
+        emit UpdateTemplarRolesHash(_templarRolesHash);
     }
 
     /**
      * @notice Nominate the given discord ID a candidate.
-     * @dev The given address must hold the NFT for the specified discord id
+     * @dev The sender must hold the NFT for the specified discord id
      */
-    function nominate(
+    function nominate(uint256 discordId) external {
```

Comment:
Just confirming:
1. Only a self-nomination is permitted? (Which is fine, just for my understanding)
2. Anyone that wants to be nominated needs to transact directly on the L2 - no relayer for this.

Also imo we don't need the separate internal function as it just means more gas for no benefit.

Reply:
Tidied up. The bot account will have the `CAN_NOMINATE` role on the EldersElection.sol contract, and the `CAN_ASSIGN` role on the Templars.sol NFT.